frontend/packages/ceph-storage-plugin/src/components/modals/clone-pvc-modal/clone-pvc-modal.tsx

```diff
@@ -0,0 +1,174 @@
+  convertToBaseValue,
+  humanizeBinaryBytes,
+  LoadingInline,
+  ResourceIcon,
+  withHandlePromise,
+} from '@console/internal/components/utils/index';
+import {
+  createModalLauncher,
+  ModalBody,
+  ModalSubmitFooter,
+  ModalTitle,
+  ModalComponentProps,
+} from '@console/internal/components/factory';
+import { k8sCreate, K8sResourceKind } from '@console/internal/module/k8s';
+import { PersistentVolumeClaimModel } from '@console/internal/models/index';
+import { HandlePromiseProps } from '@console/internal/components/utils/promise-component';
+import { usePrometheusPoll } from '@console/internal/components/graphs/prometheus-poll-hook';
+import { PrometheusEndpoint } from '@console/internal/components/graphs/helpers';
+import { getInstantVectorStats } from '@console/internal/components/graphs/utils';
+import { DataPoint, PrometheusResponse } from '@console/internal/components/graphs/index';
+import { getPVCUsedCapacityQuery } from '../../../constants/queries';
+import './_clone-pvc-modal.scss';
+
+const accessModeRadios = Object.freeze({
+  ReadWriteOnce: 'Single User (RWO)',
+  ReadWriteMany: 'Shared Access (RWX)',
+  ReadOnlyMany: 'Read Only (ROX)',
+});
+
+const ClonePVCModal = withHandlePromise((props: ClonePVCModalProps) => {
+  const { close, cancel, resource, handlePromise, errorMessage, inProgress } = props;
+  const pvcName: string = _.get(resource, 'metadata.name');
+  const [buttonDisabled, setButton] = React.useState(false);
+  const [clonePVCName, setClonePVCName] = React.useState(`${pvcName}-clone`);
+
+  const query: string = getPVCUsedCapacityQuery(pvcName);
+
+  const [response, error, loading] = usePrometheusPoll({
+    endpoint: PrometheusEndpoint.QUERY,
+    query,
+  }) as [PrometheusResponse, Error, boolean];
+
+  const results: DataPoint[] = getInstantVectorStats(response, null, humanizeBinaryBytes);
+  const pvcUsedCapacity: string = _.get(results[0], 'label')
+    ? _.get(results[0], 'label')
+    : 'No Data';
+
+  const submit = (event: React.FormEvent<EventTarget>) => {
+    event.preventDefault();
+
+    const pvcCloneObj = {
+      apiVersion: 'v1',
+      kind: PersistentVolumeClaimModel.kind,
+      metadata: {
+        name: clonePVCName,
+        namespace: _.get(resource, 'metadata.namespace'),
+      },
+      spec: {
+        storageClassName: _.get(resource, 'spec.storageClassName'),
+        dataSource: {
+          name: pvcName,
+          kind: PersistentVolumeClaimModel.kind,
+          apiGroup: '',
+        },
+        accessModes: _.get(resource, 'spec.accessModes'),
+        resources: {
+          requests: {
+            storage: _.get(resource, 'spec.resources.requests.storage'),
+          },
+        },
+      },
+    };
+
+    handlePromise(k8sCreate(PersistentVolumeClaimModel, pvcCloneObj))
+      .then(() => {
+        close();
+      })
+      .catch((err) => {
+        setButton(true);
+        throw err;
+      });
+  };
+
+  const updateClonePVCName = (value: string) => {
```

Comment:
This helper just wraps `setClonePVCName` with no other changes. Just call `setClonePVCName` directly